### ISSUE TITLE
Attribute converter like functionality for spring-cloud-gcp-data-datastore

### ISSUE THREAD

**Issue by nbali** [GoogleCloudPlatform/spring-cloud-gcp#438]
**Is your feature request related to a problem? Please describe.**
I'm unable to do custom conversions of specific fields between the pojo and db state, because this widely available feature is missing from spring-cloud-gcp-data-datastore, 2.0.2

**Describe the solution you'd like**
https://www.baeldung.com/jpa-attribute-converters#using-the-converter for spring-cloud-gcp-data-datastore :)

**Describe alternatives you've considered**
Without spring data support, I'm forced to use https://github.com/objectify/objectify/blob/master/src/main/java/com/googlecode/objectify/annotation/Translate.java.

**Additional context**
-

**Comment by meltsufin** [comment]
You're right about only global converters being currently supported as described [here](https://googlecloudplatform.github.io/spring-cloud-gcp/2.0.2/reference/html/index.html#custom-types-2). If that doesn't work for you, something like JPA's `@Convert` would need to be added. Unfortunately, I don't think there is such a facility in Spring Data Commons (cc/ @mp911de).
As always, we welcome contributions.

**Comment by mp911de** [comment]
Right now, there's no infrastructure for property-specific converters. Neo4j and Elasticsearch went forward to introduce their store-specific mechanisms. We have a ticket in Spring Data Commons (https://github.com/spring-projects/spring-data-commons/issues/1484) to design and come up with some infrastructure.

**Comment by nbali** [comment]
Well this tells me the "issue" is real, but just not painful enough for anybody, so everybody is doing a workaround instead. The issue was opened 4 years ago, and barely any interaction on it. Anyway I always thought it's strange it was missing from spring-data and this was usually the reason we went with JPA instead - when we could. My guess is JPA was a too easy and quick alternative for others as well. Not an option with datastore. Unfortunately my knowledge is far from deep enough to contribute code to such a huge feature - even so that I guess there is a reason Neo4j and Elasticsearch did the specific version as well, and not the general one.

So what is the expected flow here? Let the issue be dependent on that one, or should it be closed?

**Comment by mp911de** [comment]
We don't have a final design for the API and since we expect some changes in the actual store module I'd suggest keeping this ticket open.

**Comment by zhumin8** [comment]
update: Spring Data Common has added ["support for property-specific converters"](https://github.com/spring-projects/spring-data-commons/pull/2566)  in [3.0.0-M2](https://github.com/spring-projects/spring-data-commons/releases/tag/3.0.0-M2)
